### ISSUE TITLE
Add support for the metadata service in `ignite run`

### ISSUE THREAD

**Issue by luxas** [weaveworks/ignite#77]
Firecracker has a [metadata service](https://github.com/firecracker-microvm/firecracker/blob/master/docs/mmds.md) that allows for passing arbitrary JSON data to the VM, where it client-side can be fetched from a hardcoded `169.254.169.254` IP endpoint.

We should allow the user to specify this metadata at runtime.

**Comment by patrobinson** [comment]
Wouldn't it be better to expose `firecracker.sock` say at `/var/lib/firecracker/vm/$id/firecracker.sock`?
This way you could dynamically update the metadata and access other features of the Firecracker API.

**Comment by luxas** [comment]
Yes, that is also a good idea (approved #132 :smile:), but at VM start time (IIRC), you need to set whether the MMDS should be enabled or not, and by default we have it disabled for less resource usage.
So we'd most probably like to expose some high-level knob to set the data, too

**Comment by patrobinson** [comment]
@luxas oh cool, I'll work on that next

**Comment by luxas** [comment]
Thank you @patrobinson :sparkles:! Let us know if you need any help.

**Comment by luxas** [comment]
Hi @patrobinson :wave: any progress here? Do you want help on getting this up and running?

**Comment by patrobinson** [comment]
I had a look at this but it wasn't trivial. It's not currently a focus for me sorry.

**Comment by luxas** [comment]
Ok, no problem. Would you like some smaller work item to help out on @patrobinson instead or are you too busy at the moment?

**Comment by patrobinson** [comment]
Thanks @luxas for being so forthcoming, but at this stage I'm not looking to actively contribute. That may change in the future and I'll let you know.

**Comment by luxas** [comment]
Gotcha. No problem :smile:!

**Comment by alexeldeib** [comment]
@luxas what are you thinking here? user passes in json on vm creation and that gets applied with a PUT after VM is up, or introducing an ignite CLI command to pass through arbitrary data to MMDS? Or something else?

**Comment by alexeldeib** [comment]
I started on this 😄 Would appreciate guidance though

**Comment by luxas** [comment]
Thanks! I'll comment on the PR :+1:
